notifications/management/commands/send_notifications.py

```diff
@@ -38,524 +22,53 @@ def add_arguments(self, parser):
-            ORDER BY events.start_time DESC
-        '''
-
-        cursor = connection.cursor()
-        cursor.execute(new_events, [tuple(committee_ids)])
-        columns = [c[0] for c in cursor.description]
-
-        updates = []
-
-        grouper = lambda x: x[1]
-        event_groups = sorted(cursor, key=grouper)
-
-        for slug, group in itertools.groupby(event_groups, key=grouper):
-            group = list(group)
-
-            committee = {
-                'name': group[0][0],
-                'slug': group[0][1],
-                'events': []
-            }
-
-            for row in group:
-                event = dict(zip(columns[2:], row[2:]))
-                committee['events'].append(event)
-
-            committee['events'] = sorted(committee['events'],
-                                         key=lambda x: x['start_time'],
-                                         reverse=True)
-
-            updates.append(committee)
-
-        return updates
-
-    def find_new_events(self):
-        new_events = '''
-            SELECT * FROM (
-            SELECT DISTINCT ON (event.ocd_id)
-              event.name,
-              event.start_time,
-              event.end_time,
-              event.slug,
-              event.all_day,
-              event.location_name,
-              event.slug
-            FROM councilmatic_core_event AS event
-            JOIN new_event AS new
-              ON event.ocd_id = new.ocd_id
-            WHERE event.start_time > NOW()
-            ) AS events
-            ORDER BY events.start_time
-        '''
-
-        cursor = connection.cursor()
-        cursor.execute(new_events)
-        columns = [c[0] for c in cursor.description]
-
-        return [dict(zip(columns, r)) for r in cursor]
-
-    def find_updated_events(self):
-        new_events = '''
-            SELECT * FROM (
-            SELECT DISTINCT ON (event.ocd_id)
-              event.name,
-              event.start_time,
-              event.end_time,
-              event.slug,
-              event.all_day,
-              event.location_name,
-              event.slug
-            FROM councilmatic_core_event AS event
-            JOIN change_event AS change
-              ON event.ocd_id = change.ocd_id
-            WHERE event.start_time > NOW()
-            ) AS events
-            ORDER BY events.start_time
-        '''
-
-        cursor = connection.cursor()
-        cursor.execute(new_events)
-        columns = [c[0] for c in cursor.description]
+    def handle(self, *args, **options):
+        users = NotificationsUser.objects.all()
+        for user in users:
+            if user.has_subscriptions:
+                # BillActionSubscription.get_updates() returns a list, so make
+                # sure to flatten the outer list.
+                bill_action_updates = filter(self._is_empty, chain.from_iterable([
+                    sub.get_updates() for sub in user.billactionsubscriptions.all()
+                ]))
```

Comment:
I can't tell how much of a problem it is that `BillActionSubscription.get_updates()` returns updates in a slightly different format than the rest of the subscriptions (list vs. dict). This was the easiest way to refactor the existing code without having to change the email templates at all. On the one hand the method is private so it won't be confusing to outside users; on the other hand it's a little weird for the API be inconsistent.

Reply:
i think it's ok to leave inconsistent for now, for the same pros you list. how about leaving a docstring highlighting the deviation from the standard format in the offending method? that way, we have something to go off of if it causes us trouble, but it doesn't require more changes to the templates.

Reply:
Sure, good thought!